src/bin/main.rs

```diff
@@ -136,12 +172,51 @@ fn check(args: &ArgMatches) -> HowserResult<Option<ValidationProblem>> {
     }
 }
 
+fn process_pharmacy_file(args: &ArgMatches) -> HowserResult<Vec<ValidationProblem>> {
+    let mut report: Vec<ValidationProblem> = Vec::new();
+
+    if let Some(filename) = args.value_of("pharmacy") {
+        if let Ok(pharmacy_contents) = get_file_contents(filename) {
+            if let Some(specs) = pharmacy_contents.parse::<Value>()?["Specs"].as_table() {
+                for rx_file in specs.keys() {
+                    if let Some(doc_file) = specs.get(rx_file).map_or(None, |value| value.as_str())
+                    {
+                        let validate_args = vec!["howser", "validate", rx_file, doc_file];
+                        let app = make_app();
+                        let matches = app.get_matches_from(validate_args);
+                        if let Some(sub_m) = matches.subcommand_matches("validate") {
+                            let mut problems = validate(sub_m)?;
+                            if args.is_present("fail-early") && !problems.is_empty() {
+                                return Ok(problems);
```

Comment:
The nesting has gotten quite deep here, you may want to consider method chaining using the likes of `ok_or` and `and_then` and `or` to flatten the flow.